src/config/api/api.types.ts

```diff
@@ -0,0 +1,24 @@
+export interface RequestParams<T> {
+  searchText?: string;
+  page?: number;
+  pageSize?: number;
+  sort?: string;
+  filter?: T;
+  with?: string | string[];
+}
+
+export interface ServicePagination {
+  count: number;
+  currentPage: number;
+  links: { next: string; previous: string };
+  perPage: number;
+  total: number;
+  totalPages: number;
+}
+
+export interface ServiceResponse<T> {
+  data: T;
```

Comment:
I'd suggest changing this to `payload` instead of `data` because `data` is already used by the axios instance. This way in our api calls instead of having `response.data.data` we would have `response.data.payload`.

Reply:
While I agree that `data` isn't the best because it's weird having `response.data.data`, this interface represents the structure the https://github.com/flugg/laravel-responder package provides (which we use for the responses). I would submit this type of change to a poll because it would involve updating the backend boilerplate so that it changes this key to what we decide.

I'm afraid I have to disagree with the use of `payload` refers to data sent by the client to the server, not something you get from the server to the client. We would have to think of something else.

Reply:
What do you think about this guys?
cc: @JLin-99 @guillermo-lightit @StokJoaco @cesarlightit @gianfranco-rocco @thomasmcculloch2 @ezef @cgarcia-lightit @UrielRadzyminski1

Reply:
I totally agree that payload is more semantic and avoids redundancy in names 👍 
Data doesn't explain the context. Payload does.

But I can live with data if backend team refuses to that.

Reply:
I agree with @gianfranco-rocco that the naming `payload` might not be the best choice, as he mentioned, it's a term typically used when the client sends data to the server and no the other way. So I would prefer to keep with `data`

What I find strange here (**and I think that's where the problem lies**) is that the `data` field contains data from the `response`, when in reality it should only contain the `data` itself.

> ![image](https://github.com/user-attachments/assets/8b32bfdb-f574-4ad8-8f7b-15256e9ac758)

but we are duplicating and getting those response information inside `data`... that's why we're using `response.data.data` instead of `response.data` directly or `response.data.pagination` instead of `response.pagination`

> <img width="882" alt="Screenshot 2024-09-20 at 2 01 18 PM" src="https://github.com/user-attachments/assets/0eec9156-7f0a-478b-a258-63b0d21966a6">

> <img width="929" alt="Screenshot 2024-09-20 at 2 01 34 PM" src="https://github.com/user-attachments/assets/0dd965be-a237-4403-be4c-45008269484d">

Ideally, the `data` field should only contain the actual `data` itself, without any additional response information.

Example from other app's GET response:
> ![image](https://github.com/user-attachments/assets/08040335-fd8a-4cc3-8f9b-11208a62880b)

Example from our app's GET response:
> ![image](https://github.com/user-attachments/assets/c7251812-de35-41c4-bd12-579a9f927805)

Reply:
@JLin-99, the reason for that structure is the package I mentioned. All Laravel projects implement it because that's the standard for our endpoint responses. It follows the [JSend specification](https://github.com/omniti-labs/jsend).

Reply:
The way I see it we have three choices here:

1. Write a custom serializer for Laravel Responder that replaces the `data` key for something else.
2. Write a custom interceptor for Axios  that replaces the `data` key for something else.
3. Do nothing and keep `response.data.data` as our standard.

My preferred option would be number one but I'm fine with all three of them to be honest.

Reply:
+1 on **not** using `payload` as the nomenclature for a service response, since its typically used for, well, body payloads (requests).

Reply:
In my opinion I prefer option 2 or 3 from Walter list, but in case we do the interceptor, I'm not sure how we should name the data (+1 not using payload). So in general, I'm ok with data.data tbh

Reply:
If we decide to change the standard, the 1st option is the best IMO.